packages/flutter_localizations/lib/src/cupertino_localizations.dart

```diff
@@ -0,0 +1,432 @@
+  ///
+  /// ## Sample code
+  ///
+  /// To include the localizations provided by this class and by
+  /// [GlobalWidgetsLocalizations] in a [CupertinoApp],
+  /// use [GlobalCupertinoLocalizations.delegates] as the value of
+  /// [CupertinoApp.localizationsDelegates], and specify the locales your
+  /// app supports with [CupertinoApp.supportedLocales]:
+  ///
+  /// ```dart
+  /// new CupertinoApp(
+  ///   localizationsDelegates: GlobalCupertinoLocalizations.delegates,
+  ///   supportedLocales: [
+  ///     const Locale('en', 'US'), // English
+  ///     const Locale('he', 'IL'), // Hebrew
+  ///   ],
+  ///   // ...
+  /// )
+  /// ```
+  static const List<LocalizationsDelegate<dynamic>> delegates = <LocalizationsDelegate<dynamic>>[
+    GlobalCupertinoLocalizations.delegate,
+    GlobalWidgetsLocalizations.delegate,
+  ];
+}
+
+class _GlobalCupertinoLocalizationsDelegate extends LocalizationsDelegate<CupertinoLocalizations> {
+  const _GlobalCupertinoLocalizationsDelegate();
+
+  @override
+  bool isSupported(Locale locale) => false; // TODO(xster): implement.
+
+  static final Map<Locale, Future<CupertinoLocalizations>> _loadedTranslations = <Locale, Future<CupertinoLocalizations>>{};
+
+  @override
+  Future<CupertinoLocalizations> load(Locale locale) {
+    assert(isSupported(locale)); // TODO(xster): implement.
+    return _loadedTranslations.putIfAbsent(locale, () {
+      util.loadDateIntlDataIfNotLoaded();
+
+      final String localeName = intl.Intl.canonicalizedLocale(locale.toString());
+
+      intl.DateFormat fullYearFormat;
+      intl.DateFormat dayFormat;
+      intl.DateFormat mediumDateFormat;
+      // We don't want any additional decoration here. The am/pm is handled in
+      // the date picker. We just want an hour number localized.
+      intl.DateFormat singleDigitHourFormat;
+      intl.DateFormat singleDigitMinuteFormat;
+      intl.DateFormat doubleDigitMinuteFormat;
+      intl.DateFormat singleDigitSecondFormat;
+      intl.NumberFormat decimalFormat;
+
+      void loadFormats(String locale) {
+        fullYearFormat = intl.DateFormat.y(locale);
+        dayFormat = intl.DateFormat.d(locale);
+        mediumDateFormat = intl.DateFormat.MMMEd(locale);
+        // TODO(xster): fix when https://github.com/dart-lang/intl/issues/207 is resolved.
+        singleDigitHourFormat = intl.DateFormat('HH', locale);
+        singleDigitMinuteFormat = intl.DateFormat.m(locale);
+        doubleDigitMinuteFormat = intl.DateFormat('mm', locale);
+        singleDigitSecondFormat = intl.DateFormat.s(locale);
+        decimalFormat = intl.NumberFormat(locale);
+      }
+
+      if (intl.DateFormat.localeExists(localeName)) {
+        loadFormats(localeName);
+      } else if (intl.DateFormat.localeExists(locale.languageCode)) {
+        loadFormats(locale.languageCode);
+      } else {
+        loadFormats(null);
+      }
+
+      assert(locale.toString() == localeName, 'comparing "$locale" to "$localeName"');
```

Comment:
The failure message doesn't seem helpful (here or in material_localizations.dart). It seems like this assertion should be right after we assign localeName.

Reply:
I'm not sure I understand what this intent is here to write the right error message. What is correction step for us or for the user?

Reply:
I guess the idea is to detect locales with oddball language or country codes like Locale('strange', 'brew'). I don't know what would be more helpful; maybe just say: Flutter doesn't support  "$locale" (canonical form is "$localeName").  Hopefully it will never happen :-)

Reply:
ok, I've made an attempt